app/src/main/java/org/wikipedia/userprofile/ContributionsFragment.kt

```diff
@@ -195,22 +193,38 @@ class ContributionsFragment : Fragment(), ContributionsHeaderView.Callback {
     }
 
     private fun homeSiteObservable(): Observable<Pair<List<Contribution>, Int>> {
-        return if (allContributions.isNotEmpty()) Observable.just(Pair(Collections.emptyList(), -1))
-        else ServiceFactory.get(WikipediaApp.getInstance().wikiSite).userInfo
+        return if (allContributions.isNotEmpty() && !continuations.containsKey(WikipediaApp.getInstance().wikiSite)) Observable.just(Pair(Collections.emptyList(), -1))
+        else ServiceFactory.get(WikipediaApp.getInstance().wikiSite).getUserContributions(AccountUtil.userName!!, 50, continuations[WikipediaApp.getInstance().wikiSite])
             .subscribeOn(Schedulers.io())
             .flatMap { response ->
-                Observable.just(Pair(emptyList(), response.query?.userInfo()!!.editCount))
+                val contributions = mutableListOf<Contribution>()
+                val cont = response.continuation["uccontinue"]
+                if (cont.isNullOrEmpty()) {
+                    continuations.remove(WikipediaApp.getInstance().wikiSite)
+                } else {
+                    continuations[WikipediaApp.getInstance().wikiSite] = cont
+                }
+                response.query?.userContributions()?.forEach {
+                    contributions.add(Contribution("", it.revid, it.title, it.title, it.title, EDIT_TYPE_GENERIC, null, it.date(),
+                        WikipediaApp.getInstance().wikiSite, 0, it.sizediff, it.top, 0))
+                }
```

Comment:
Looks like the order of all the contributions is showing incorrectly (not ordered by date).

Reply:
@cooltey  I can't reproduce. We sort all the contributions by date in the `createConsolidatedList()` function when building the list to be displayed.  Are you totally sure you saw your contributions out of order? Can you post a screenshot?

Reply:
Ah! My bad. I guess somehow I was confused by Jun and Jul. The order looks good to me now.

However, I am not sure about the edits streak. In my task screen it shows 5days edit streak, but in the contribution screen it doesn't look like I have a 5 days edit streak.

![Screenshot_20210803-125939_Wikipedia Dev.jpg](https://user-images.githubusercontent.com/2435576/128078533-3144ce22-427a-44e1-9094-d503a0878ad1.jpg)

![Screenshot_20210803-125928_Wikipedia Dev.jpg](https://user-images.githubusercontent.com/2435576/128078544-aa1858ae-c218-4924-af6c-8eac26ffb8fe.jpg)

Reply:
Well, it looks like the streak logic in `getEditStreak()` is just wrong and needs to be fixed. (i.e. not related to this PR)

Reply:
@cooltey I made a quick fix that should improve the edit streak logic, although it's still naive and not expected to be perfect.